### PR TITLE
[User management] Add feature flag to enable user management [DPP-827]

### PR DESCRIPTION
**Pull request by pbatko-da**
CHANGELOG_BEGIN
Sandbox: Participant server now supports '--feature-user-management' flag to turn on or off user management feature.
CHANGELOG_END

### CODE REVIEW

File: ledger/sandbox-common/src/main/scala/platform/sandbox/config/SandboxConfig.scala [hunk 1]
```diff
@@ -176,7 +176,7 @@ object SandboxConfig {
       sqlStartMode = Some(DefaultSqlStartupMode),
       enableCompression = false,
       enableSelfServiceErrorCodes = true,
-      userManagementConfig = UserManagementConfig.default,
+      userManagementConfig = UserManagementConfig.default(true),
```
Comment (mziolekda) [inline ledger/sandbox-common/src/main/scala/platform/sandbox/config/SandboxConfig.scala:179]: Can you please remind me why the default is different between kvutils and sandboxes?

Comment (pbatko-da) [reply ledger/sandbox-common/src/main/scala/platform/sandbox/config/SandboxConfig.scala:179]: Talked offline

**Review by mziolekda — APPROVED** [review]

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala [hunk 2]
```diff
@@ -652,10 +652,19 @@ object Config {
           )
           .action((_, config: Config[Extra]) => config.copy(enableSelfServiceErrorCodes = false))
 
+        opt[Boolean]("feature-user-management")
```
[the remark this answers is not in the document]
Comment (pbatko-da) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:655]: @meiersi-da @mziolekda what's your view?

[the remark this answers is not in the document]
Comment (mziolekda) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:655]: Eagle eye! It was discussed in the past and the design document calls for `--enable-user-management`

[the remark this answers is not in the document]
Comment (pbatko-da) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:655]: fyi the jira story mentions "feature-user-management" https://digitalasset.atlassian.net/browse/DPP-827

[the remark this answers is not in the document]
Comment (meiersi-da) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala:655]: AFAIR, we've agreed on `--enable-user-management`. The JIRA story is out of date.

[the remark this answers is not in the document]
Comment (pbatko-da) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala]: Change to `--enable-user-management` applied

**Pull request merged**